### PR TITLE
Feature/didpeer4

### PR DESCRIPTION
**Pull request by Patrik-Stas**
Implement `did:peer:4` ( see https://identity.foundation/peer-did-method-spec/#method-4-short-form-and-long-form ) as `PeerDid<Numalgo4>` 

#### `DidPeer4EncodedDocument` 
- Comes together with builder `DidPeer4EncodedDocumentBuilder`.  The `DidPeer4EncodedDocument` is a structure very similar to Did Document, but with small differences. It's a structure which according to spec, you start with in order to construct new `did:peer:4`. The differences of this structure, in comparison to standard W3C Did Document are specified here: https://identity.foundation/peer-did-method-spec/#creating-a-did
#### `PeerDid<Numalgo4>::new(encoded_document: DidPeer4EncodedDocument) - Result<Self, DidPeerError>` 
- given instance of `DidPeer4EncodedDocument` (see previous point) this function turns it into PeerDid structure.

#### `PeerDid<Numalgo4>::long_form(&self) -> Result<Did, DidPeerError>` 
- returns long form of peer did 

#### `PeerDid<Numalgo4>::short_form(&self) -> Result<Did, DidPeerError>` 
- returns short form of peer did  

#### `PeerDid<Numalgo4>::hash(&self)  -> Result<String, DidPeerError>` 
- returns multibase hash of encoded pseudo-did-document

#### `PeerDid<Numalgo4>::resolve_did_doc(&self) -> Result<DidDocument, DidPeerError>` 
- resolves instance of did:peer:4 to a Did Document.

### CODE REVIEW

File: did_core/did_methods/did_peer/src/peer_did/numalgos/mod.rs [hunk 1]
```diff
@@ -21,12 +22,12 @@ pub trait Numalgo: Sized + Default {
         Did: TryFrom<T>,
         <Did as TryFrom<T>>::Error: Into<DidPeerError>,
     {
+        println!("parsing did inside parse<T> function in numalgo.rs");
```
Comment (gmulhearn) [inline did_core/did_methods/did_peer/src/peer_did/numalgos/mod.rs:25]: nit: println

File: did_core/did_methods/did_peer/src/peer_did/numalgos/numalgo4/mod.rs [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+    error::DidPeerError,
+    peer_did::{
+        numalgos::{numalgo4::encoded_document::DidPeer4EncodedDocument, Numalgo},
+        PeerDid,
+    },
+};
+
+pub mod encoded_document;
+
+// // The document MUST NOT include an id at the root. For DID Documents, this is populated with the
```
Comment (gmulhearn) [inline did_core/did_methods/did_peer/src/peer_did/numalgos/numalgo4/mod.rs:14]: nit: this comment block seems a bit random. is it left-over?

**Review by gmulhearn — DISMISSED** [review]
nice! just a couple nit comments above

**Comment by Patrik-Stas** [comment]
Due github issues I re-opened the pull request separately again: AbsaOSS:feature/didpeer4

Closing this one

**Pull request closed without merging**